include/revng/Yield/Internal/Instruction.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <limits>
+#include <string>
+
+#include "revng/ADT/SortedVector.h"
+#include "revng/Model/VerifyHelper.h"
+#include "revng/Support/MetaAddress.h"
+#include "revng/Support/MetaAddress/YAMLTraits.h"
+#include "revng/Yield/Internal/TagType.h"
+
+namespace yield {
+using ByteContainer = llvm::SmallVector<uint8_t, 16>;
+}
+
+/* TUPLE-TREE-YAML
+
+name: Instruction
+type: struct
+fields:
+  - name: Address
+    type: MetaAddress
+  - name: Bytes
+    type: yield::ByteContainer
+  - name: Raw
+    type: std::string
+
+  - name: Tags
+    sequence:
+      type: std::vector
```

Comment:
* Consider using `SortedVector` and make `yield::Tag` with a key spanning all the fields of the data structure? It can work as a set, avoiding issues with duplicate tags.
* I'm afraid `mnemonic` +  tag set will inflate in a more and more sophisticated data structure and I really wonder if we should avoid representing the structure here entirely and just switch to XML at this stage, i.e., drop mnemonic and tags and stuff XML into the `Raw`/`Disassembly` field. We're going from 1) pseudo-XML from LLVM disassembler, 2) to out-of-line markup here, 3) back to XML when we'll ship this to the client and finally 4) in VSCode we'll translate it back to out-of-line markup again. Maybe we can avoid 2) and go from 1) to 3)?

Reply:
I feel like this way to store it gives the most flexibility possible. If at some point we decide to change the approach, there'll be a lot less work to do with having tags and the raw text separated.

Reply:
OK, let's stick with this.

I'm closing this discussion in favor of the other one where you mentioned `<=>`.